modules/gdnative/videodecoder/video_stream_gdnative.h

```diff
@@ -0,0 +1,209 @@
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef VIDEO_STREAM_FFMPEG_H
+#define VIDEO_STREAM_FFMPEG_H
+
+#include <modules/gdnative/gdnative.h>
+#include <os/file_access.h>
+#include <scene/resources/texture.h>
+#include <scene/resources/video_stream.h>
+
+struct VideoDecoderGDNative {
+	String plugin_name;
+	Vector<String> supported_extensions;
+	const godot_videodecoder_interface_gdnative *interface;
+
+	VideoDecoderGDNative() :
+			interface(NULL),
+			plugin_name("none") {}
+
+	VideoDecoderGDNative(const godot_videodecoder_interface_gdnative *p_interface) :
+			interface(p_interface),
+			plugin_name(p_interface->get_plugin_name()) {
+		_get_supported_extensions();
+	}
+
+	void set_interface(const godot_videodecoder_interface_gdnative *p_interface) {
+		interface = p_interface;
+		plugin_name = p_interface->get_plugin_name();
+		_get_supported_extensions();
+	}
+
+private:
+	void _get_supported_extensions() {
+		supported_extensions.clear();
+		int num_ext;
+		char **supported_ext = interface->get_supported_extensions(&num_ext);
+		for (int i = 0; i < num_ext; i++) {
+			supported_extensions.push_back(supported_ext[i]);
+		}
+	}
+};
+
+class VideoDecoderServer {
+private:
+	Vector<VideoDecoderGDNative *> decoders;
+	Map<String, int> extensions;
+
+	static VideoDecoderServer *instance;
+
+public:
+	static VideoDecoderServer *get_instance() {
+		return instance;
+	}
+
+	const Map<String, int> &get_extensions() {
+		return extensions;
+	}
+
+	void register_decoder_interface(const godot_videodecoder_interface_gdnative *p_interface) {
+		VideoDecoderGDNative *decoder = memnew(VideoDecoderGDNative);
+		decoder->set_interface(p_interface);
+		int index = decoders.size();
+		printf("RDI\n");
```

Comment:
Same here, if you have to, use `print_line` :)

Reply:
boop

Reply:
This had been changed before push. My local has no printf's in the folder. I do not understand why this is showing up for you.